### DISCUSSION TITLE
Add option to redirect the base mealie url to the global default group

### DISCUSSION THREAD

**Discussion by nkringle** [mealie-recipes/mealie#2751] [Feature Request]
### First Check

- [X] I used the GitHub search to find a similar requests and didn't find it.
- [X] Checked the [tasks tagged](https://github.com/hay-kot/mealie/issues?q=is%3Aissue+is%3Aopen+label%3Atask+) issues and verified my feature is not covered

### Please provide a concise description of the problem that would be addressed by this feature.

I'm looking to replicate the 0.5 mealie functionality of not requiring a user to be logged in to view recipes. While technically you do not need to be authenticated to access the /g/home URL, right now a user would not be able to get to it without a bookmark or something - which isn't super user-friendly. Basically, I want to share my mealier URL (cookbook.someurl.com) with friends and family without needing to give them a log in or tell them to go to cookbook.someurl.com/g/home.		

### Please provide a concise description of the feature that would resolve your issue.

I was able to POC this by using my (external to mealie) load balancer to redirect `cookbook.someurl.com\` to `cookbook.someurl.com\g\home` and not match on any other URLs. In testing, the API and login flow all worked as expected. However, for most users this would probably make sense to be a group setting or environment variable to default users to a log-in page OR a specific group public page?

I'd guess part of the redirect logic would [live here](https://github.com/mealie-recipes/mealie/blob/e8db1bbe8032e61feef4043ebc26df87f155b3f2/frontend/pages/index.vue#L16-L17).

### Please consider and list out some caveats or tradeoffs made in your design decision

**Where does this setting live?** 
It is a setting that would apply without the context of a specific user, so I'd imagine it needs to be applied at the global to mealie level. Maybe that's an environment variable? Not sure. Is there a concept of a default group and if so could you just use that?

**What is the setting called?**
I could see it being something like 'Redirect Non-Logged-In Users to a public group` and specific the group name (ie, `home`). If off, redirect / to `/login` as it is being done today. There is also going to be some level of documentation to indicate that the group you are redirecting to must have public access enabled.  

### Additional Information

- [ ] If this is accepted I'm willing to submit a PR to provide this feature
- [ ] If this is accepted I'm willing to help maintain this feature
- [ ] I'm willing to sponsor/pay a developer to do this work

**Comment by Skrymir** [comment]
I came looking for this very thing.  Having the base URL redirect to publicly shared recipes insteast of /login is a great idea.  In old versions of mealie I would only login to add or edit.
